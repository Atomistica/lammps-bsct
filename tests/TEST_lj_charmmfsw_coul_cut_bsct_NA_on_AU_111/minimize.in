# Modify and minimize

variable hvac     index 30.0 # desired height of vacuum
variable dz       index 15.0 # desired distance between ion and substrate
variable ion_type index 2    # type of ion (Na)

units           real

neigh_modify    delay 0 every 1 check yes

atom_style      full
bond_style      harmonic
angle_style     charmm
dihedral_style  charmmfsw
improper_style  harmonic

pair_style      lj/charmmfsw/coul/charmmfsh/bsct 8.0 12.0 9.0

special_bonds   charmm

read_data       relaxed.lammps

print           "Initial xlo xhi ylo yhi zlo zhi:"
print           "  $((xlo)) $((xhi))"
print           "  $((ylo)) $((yhi))"
print           "  $((zlo)) $((zhi))" 

# surface z coordinate
variable        z0 equal zhi

# intended position of ion:
variable        x1 equal $((xhi-xlo)/2.0)
variable        y1 equal $((yhi-ylo)/2.0)
variable        z1 equal $(v_z0 + v_dz)

change_box      all z final $((zlo)) $((zhi+v_hvac))

print           "Resized xlo xhi ylo yhi zlo zhi:"
print           "  $((xlo)) $((xhi))"
print           "  $((ylo)) $((yhi))"
print           "  $((zlo)) $((zhi))" 

group           surface          type  1     #  1 - Au
group           ion              type  2     #  2 - Na+
group           bsct_atoms       union ion surface

thermo          1
thermo_style    custom step temp press etotal ke pe &
                ebond eangle edihed eimp epair evdwl ecoul elong etail
thermo_modify   format float %15.14g norm no

##############
# minimization
##############

### Minimze the potential energy here.
### minimize energy_tol force_tol maxiter maxeval
minimize        0.0 1.0e-6 1000 1000
# Require a force tolerance of ~ 10 Kcal/mole-Angstrom
# That should correspond to about 400 kJ / (mol*nm)

# Quote from http://lammps.sandia.gov/doc/minimize.html
# Perform an energy minimization of the system, by iteratively adjusting atom
# coordinates. Iterations are terminated when one of the stopping criteria is
# satisfied. At that point the configuration will hopefully be in local
# potential energy minimum. More precisely, the configuration should
# approximate a critical point for the objective function (see below), which
# may or may not be a local minimum.
#
#    etol = stopping tolerance for energy (unitless)
#    ftol = stopping tolerance for force (force units)
#    maxiter = max iterations of minimizer
#    maxeval = max number of force/energy evaluations
#
# Either or both of the etol and ftol values can be set to 0.0, in which case
# some other criterion will terminate the minimization.

print           "Insert ion at (x,y,z) = ( $((v_x1)), $((v_y1)), $((v_z1)) )"
create_atoms    ${ion_type} single $((v_x1)) $((v_y1)) $((v_z1))

write_data      minimized.lammps
write_dump      all image minimized.png type type modify backcolor white