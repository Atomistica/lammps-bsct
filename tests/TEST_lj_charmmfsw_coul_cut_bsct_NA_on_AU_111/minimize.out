LAMMPS (7 Aug 2019)
  using 1 OpenMP thread(s) per MPI task
Switching to CHARMM coulomb energy conversion constant (../pair_lj_charmmfsw_coul_charmmfsh.cpp:54)
Reading data file ...
  orthogonal box = (0.956923 0.55248 1.18849) to (29.6646 30.3864 29.3162)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  1440 atoms
  reading velocities ...
  1440 velocities
Finding 1-2 1-3 1-4 neighbors ...
  special bond factors lj:   0          0          0         
  special bond factors coul: 0          0          0         
  0 = max # of 1-2 neighbors
  0 = max # of 1-3 neighbors
  0 = max # of 1-4 neighbors
  1 = max # of special neighbors
  special bonds CPU = 0.000372576 secs
  read_data CPU = 0.00556611 secs
Initial xlo xhi ylo yhi zlo zhi:
  0.95692267392453156383 29.664602891660422301
  0.5524796001652361177 30.386378009088105046
  1.1884930960675590406 29.316163036333460212
Changing box ...
  orthogonal box = (0.956923 0.55248 1.18849) to (29.6646 30.3864 59.3162)
Resized xlo xhi ylo yhi zlo zhi:
  0.95692267392453156383 29.664602891660422301
  0.5524796001652361177 30.386378009088105046
  1.1884930960675590406 59.316163036333463765
1440 atoms in group surface
0 atoms in group ion
1440 atoms in group bsct_atoms
Neighbor list info ...
  update every 1 steps, delay 0 steps, check yes
  max neighbors/atom: 2000, page size: 100000
  master list distance cutoff = 14
  ghost atom cutoff = 14
  binsize = 7, bins = 5 5 9
  1 neighbor lists, perpetual/occasional/extra = 1 0 0
  (1) pair lj/charmmfsw/coul/charmmfsh/bsct, perpetual
      attributes: half, newton on
      pair build: half/bin/newton
      stencil: half/bin/3d/newton
      bin: standard
Setting up cg style minimization ...
  Unit style    : real
  Current step  : 0
Per MPI rank memory allocation (min/avg/max) = 8.924 | 8.924 | 8.924 Mbytes
Step Temp Press TotEng KinEng PotEng E_bond E_angle E_dihed E_impro E_pair E_vdwl E_coul E_long E_tail 
       0               0 2838.2495851651 -59490.861560515               0 -59490.861560515               0               0               0               0 -59490.861560515 -59490.861560515               0               0               0 
       1               0 2201.3372399405 -59497.15908533               0 -59497.15908533               0               0               0               0 -59497.15908533 -59497.15908533               0               0               0 
       2               0 1986.6036831256 -59497.888006717               0 -59497.888006717               0               0               0               0 -59497.888006717 -59497.888006717               0               0               0 
       3               0 2041.4805184999 -59499.845462931               0 -59499.845462931               0               0               0               0 -59499.845462931 -59499.845462931               0               0               0 
       4               0 1994.8014435192 -59500.054831328               0 -59500.054831328               0               0               0               0 -59500.054831328 -59500.054831328               0               0               0 
       5               0 1941.9664259345 -59500.12396076               0 -59500.12396076               0               0               0               0 -59500.12396076 -59500.12396076               0               0               0 
       6               0 1913.9397417363 -59500.275477236               0 -59500.275477236               0               0               0               0 -59500.275477236 -59500.275477236               0               0               0 
       7               0 1849.7310305518 -59500.31605563               0 -59500.31605563               0               0               0               0 -59500.31605563 -59500.31605563               0               0               0 
       8               0 1770.9136709731 -59500.343721722               0 -59500.343721722               0               0               0               0 -59500.343721722 -59500.343721722               0               0               0 
       9               0  1679.411711742 -59500.411969903               0 -59500.411969903               0               0               0               0 -59500.411969903 -59500.411969903               0               0               0 
      10               0 1552.9285683842 -59500.482251589               0 -59500.482251589               0               0               0               0 -59500.482251589 -59500.482251589               0               0               0 
      11               0 1486.3253366216 -59500.507808444               0 -59500.507808444               0               0               0               0 -59500.507808444 -59500.507808444               0               0               0 
      12               0 1446.9945821362 -59500.530621707               0 -59500.530621707               0               0               0               0 -59500.530621707 -59500.530621707               0               0               0 
      13               0 1446.3451247504 -59500.532003246               0 -59500.532003246               0               0               0               0 -59500.532003246 -59500.532003246               0               0               0 
      14               0 1451.5070890863 -59500.533579471               0 -59500.533579471               0               0               0               0 -59500.533579471 -59500.533579471               0               0               0 
      15               0 1452.1614759537 -59500.533619832               0 -59500.533619832               0               0               0               0 -59500.533619832 -59500.533619832               0               0               0 
      16               0  1452.973145513 -59500.533626224               0 -59500.533626224               0               0               0               0 -59500.533626224 -59500.533626224               0               0               0 
      17               0  1452.973145513 -59500.533626224               0 -59500.533626224               0               0               0               0 -59500.533626224 -59500.533626224               0               0               0 
Loop time of 0.760566 on 1 procs for 17 steps with 1440 atoms

100.0% CPU use with 1 MPI tasks x 1 OpenMP threads

Minimization stats:
  Stopping criterion = linesearch alpha is zero
  Energy initial, next-to-last, final = 
        -59490.8615605     -59500.5336262     -59500.5336262
  Force two-norm initial, final = 68.8735 0.0604054
  Force max component initial, final = 3.31181 0.0034109
  Final line search alpha, max atom move = 1.90735e-06 6.50577e-09
  Iterations, force evaluations = 17 71

MPI task timing breakdown:
Section |  min time  |  avg time  |  max time  |%varavg| %total
---------------------------------------------------------------
Pair    | 0.7554     | 0.7554     | 0.7554     |   0.0 | 99.32
Bond    | 3.9451e-05 | 3.9451e-05 | 3.9451e-05 |   0.0 |  0.01
Neigh   | 0          | 0          | 0          |   0.0 |  0.00
Comm    | 0.0024139  | 0.0024139  | 0.0024139  |   0.0 |  0.32
Output  | 0.00044664 | 0.00044664 | 0.00044664 |   0.0 |  0.06
Modify  | 0          | 0          | 0          |   0.0 |  0.00
Other   |            | 0.00227    |            |       |  0.30

Nlocal:    1440 ave 1440 max 1440 min
Histogram: 1 0 0 0 0 0 0 0 0 0
Nghost:    6960 ave 6960 max 6960 min
Histogram: 1 0 0 0 0 0 0 0 0 0
Neighs:    396000 ave 396000 max 396000 min
Histogram: 1 0 0 0 0 0 0 0 0 0

Total # of neighbors = 396000
Ave neighs/atom = 275
Ave special neighs/atom = 0
Neighbor list builds = 0
Dangerous builds = 0
Insert ion at (x,y,z) = ( 14.353840108867899517, 14.916949204461399603, 44.316163036333499292 )
Created 1 atoms
  create_atoms CPU = 2.6135e-05 secs
System init for write_data ...
Total wall time: 0:00:00
